### PR TITLE
Add pkg-config file generation

### PR DESCRIPTION
**Pull request by karczex**
FIX: #5

Signed-off-by: Pawel Karczewski <pawel.karczewski@intel.com>

### CODE REVIEW

**Review by AlexVeprev — COMMENTED** [review]
@karczex, thank you for the PR!

Is it enough for you to have `pkg-config` support for build from sources only? Are you interested in having it in prebuilt packages from [release](https://github.com/oneapi-src/oneTBB/releases) page (not sure how to dynamically get `prefix` value in this case)?

Overall comment: the PR covers only `tbb` library in release configuration, but we need to take into account other oneTBB libraries (tbbmalloc, tbbbind, etc), consider debug libraries that have `_debug` suffix (`-ltbb_debug`). Most likely we should implement a common macro to run it for each target.

**Comment by karczex** [comment]
It would be great to have pkg-config for prebuilt, but I'm not aware of any way to parametrise prefix in runtime. I consider, having this feature only for builds from sources is good enougth.

**Comment by KFilipek** [comment]
AFAIK in distro packages are also *.pc files, so I believe it's enough to have them only in sources and deb/rpm files. 
For prebuilt packages I see one advantage of having default location in .pc file, you can copy the latest release of TBB to default directory /usr (e.g. on older distro when you don't have access to newer package versions) and use it like normally installed, it's a cornercase.
Approach in this PR cover most real world cases.

**Comment by jtojnar** [comment]
pkg-config file is now part of the repo. Introduced in the following megacommit https://github.com/oneapi-src/oneTBB/commit/478de5b1887c928e52f029d706af6ea640a877be#diff-bd12f4ac9cbdca8ed4ea6a1a5e12e550ba667de6ea9bf42a8a1718704af4a80c

File: tbb.pc.in [hunk 1]
```diff
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name: oneAPI Threading Building Blocks
+Description: Intel's parallelism library for C++
+URL: https://github.com/oneapi-src/oneTBB
+Version: @TBB_VERSION@
+Libs: -L${libdir} -ltbb -latomic
```
[the remark this answers is not in the document]
Comment (KoyamaSohei) [reply tbb.pc.in:10]: Is `-latomic` required?
I did research that, but I wasn't sure when we need `-latomic`. (Probably depends on the environment)



`-latomic` come from pre-built package (e.g.  [libtbb-dev/focal 2020.1-2 amd64](https://packages.ubuntu.com/en/focal/amd64/libtbb-dev) )

```
$ apt list --installed | grep tbb
libtbb-dev/focal,now 2020.1-2 amd64
libtbb2/focal,now 2020.1-2 amd64
$ cat /usr/lib/x86_64-linux-gnu/pkgconfig/tbb.pc
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib/x86_64-linux-gnu
includedir=${prefix}/include

Name: Threading Building Blocks
Description: Intel's parallelism library for C++
URL: http://www.threadingbuildingblocks.org/
Version: 
Libs: -ltbb -latomic
Cflags:
```

However, some packages did not require `-latomic`.

```
$ yum list | grep tbb
tbb.x86_64                                 4.1-9.20130314.el7     @base         
tbb-devel.x86_64                           4.1-9.20130314.el7     @base         
tbb.i686                                   4.1-9.20130314.el7     base          
tbb-debuginfo.x86_64                       4.1-9.20130314.el7     base-debuginfo
tbb-devel.i686                             4.1-9.20130314.el7     base          
tbb-doc.x86_64                             4.1-9.20130314.el7     base
$ cat /usr/lib64/pkgconfig/tbb.pc
Name: Threading Building Blocks
Description: The Threading Building Blocks library abstracts low-level threading details
URL: http://www.threadingbuildingblocks.org/
Version: 4.1.3
Libs: -ltbb
```

Thanks.

**Comment by karczex** [comment]
This PR seems to be no longer needed as pkg-config support was introduced as part of bigger patch. Thanks.

**Pull request closed without merging**